core/templates/dev/head/pages/exploration_editor/simple_editor_tab/questions/ItemSelectionEditorDirective.js

```diff
@@ -0,0 +1,305 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for a checkbox question in the simple editor.
+ */
+
+// NOTE TO DEVELOPERS: This is meant to be a reusable directive, so its only
+// dependencies should be standard utility services. It should not have any
+// concept of "state in an exploration".
+oppia.directive('itemSelectionEditor', [
```

Comment:
Shouldn't the name here be chechboxEditor and same for the file name^? It'll be more relevant for simple-editor.

Reply:
That is right, but the back end only accepts ItemSelectionInput, so for consistency's sake, I decided to keep everything the same accept what is shown to the user. We will need to create a converter in the future, but since I'm moving onto another Oppia project, it'll be up to someone else to do that.

Reply:
Actually I think you can still call this checkboxEditor. The id/name-of-directive mapping should be taken care of in StatesToQuestionsService.js.

Reply:
I did attempt at that prior to making this PR, it still doesn't seem to work...

Reply:
Hm, take a look at #3897 and #3895 -- I think they do this successfully. In what way is this interaction different from those two?

Reply:
It's because those interactions have the name name as they were before (ie. no name change), this one is represented as ItemSelection in the backend. Hence, why there weren't any issues with MC either.

Reply:
For this it's CheckboxInput for the simple editor and ItemSelectionInput for the advanced editor.

Reply:
Sorry, I still don't understand why that's an issue... The corresponding directive in #3895 is called textEditor, which doesn't match the name TextInput. So I don't see why the name of the directive matters.

What error are you getting when you use checkboxEditor instead?

Reply:
Never mind, it looks like my combo of names were not named properly.

Reply:
Fixed!